pinot-common/src/main/java/com/linkedin/pinot/common/utils/helix/HelixHelper.java

```diff
@@ -393,4 +392,16 @@ public IdealState apply(IdealState idealState) {
 
     updateIdealState(helixManager, tableName, updater, DEFAULT_RETRY_POLICY);
   }
+
+  public static List<String> getEnabledInstancesWithTag(HelixAdmin helixAdmin, String helixClusterName, String instanceTag) {
+    List<String> instancesWithTag = helixAdmin.getInstancesInClusterWithTag(helixClusterName, instanceTag);
```

Comment:
`getInstancesInClusterWithTag` fetches all instance configs, reads them and only returns the ones that have a particular tag in their list of tags. At that point, you're probably better off fetching all instance configs directly to check if it's enabled instead of fetching them twice.